Framework/Algorithms/src/PDFFourierTransform.cpp

```diff
@@ -137,6 +140,89 @@ std::map<string, string> PDFFourierTransform::validateInputs() {
   return result;
 }
 
+size_t
+PDFFourierTransform::determineQminIndex(const std::vector<double> &Q,
+                                        const std::vector<double> &FofQ) {
+  double qmin = getProperty("Qmin");
+
+  // check against available Q-range
+  if (isEmpty(qmin)) {
+    qmin = Q.front();
+  } else if (qmin < Q.front()) {
+    g_log.information(
+        "Specified Qmin < range of data. Adjusting to data range.");
+    qmin = Q.front();
+  }
+
+  // get index for the Qmin from the Q-range
+  auto q_iter = std::upper_bound(Q.begin(), Q.end(), qmin);
+  size_t qmin_index = std::distance(Q.begin(), q_iter);
+  if (qmin_index == 0)
+    qmin_index += 1; // so there doesn't have to be a check in integration loop
+
+  // go to first non-nan value
+  q_iter = std::find_if(std::next(FofQ.begin(), qmin_index), FofQ.end(),
+                        boost::math::isnormal<double>);
```

Comment:
Is this what you want? `0.0` is *not* normal. Maybe `!isfinite`? Also note that you do not need the `boost` versions, C++11 has them in `cmath`.